flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/TaskStateAssignment.java

```diff
@@ -169,31 +170,37 @@ public OperatorSubtaskState getSubtaskState(OperatorInstanceID instanceID) {
                                 instanceID,
                                 inputOperatorID,
                                 getUpstreamAssignments(),
-                                assignment ->
-                                        assignment.outputSubtaskMappings.get(
-                                                getAssignmentIndex(
-                                                        assignment.getDownstreamAssignments(),
-                                                        this)),
-                                assignment ->
-                                        assignment.getOutputMapping(
-                                                getAssignmentIndex(
-                                                        assignment.getDownstreamAssignments(),
-                                                        this)),
+                                (assignment, recompute) -> {
```

Comment:
This is definitively not better. I thought more about:

```
(assignment, recompute) ->                                                        assignment.getOutputMapping(
                                                getAssignmentIndex(
                                                        assignment.getDownstreamAssignments(),
                                                        this),
                                                        recompute)
```

Reply:
Yes, I don't like that either, as you expose additional flag to a `public` method of the class. Moreover, it's not that different from the current version. It's just that we move some part of the lambda inside of the `getOutputMapping` method.

Reply:
Well the flag expresses more the intent and you avoid passing to seemingly identical lambdas to the same function. I don't know what the issue with the `public method is` but you could add a non-public overload.